types/block.go

```diff
@@ -76,19 +78,44 @@ func (roots NmtRoots) Bytes() [][]byte {
 	return res
 }
 
-func NmtRootsFromBytes(in [][]byte) NmtRoots {
-	roots := make([]namespace.IntervalDigest, len(in))
+func NmtRootsFromBytes(in [][]byte) (roots NmtRoots, err error) {
+	roots = make([]namespace.IntervalDigest, len(in))
 	for i := 0; i < len(in); i++ {
-		roots[i] = namespace.IntervalDigestFromBytes(NamespaceSize, in[i])
+		roots[i], err = namespace.IntervalDigestFromBytes(NamespaceSize, in[i])
+		if err != nil {
+			return
+		}
+	}
+	return
+}
+
+// String returns hex representation of merkle hash of the DAHeader.
+func (dah *DataAvailabilityHeader) String() string {
+	if dah == nil {
+		return "<nil DAHeader>"
 	}
-	return roots
+	return strings.ToUpper(fmt.Sprintf("%x", dah.Hash()))
+}
+
+// IsZero checks if the DAHeader stands for Block with no shares.
+func (dah *DataAvailabilityHeader) IsZero() bool {
+	return len(dah.RowsRoots) == 0 && len(dah.ColumnRoots) == 0
 }
```

Comment:
Good question! In the code before latest John’s PR, that was possible

Reply:
I don't see why it can't be zero? If there are literally no transactions in two blocks in a row, then there are no txs, no ISRs, no evidence, and no messages.

Reply:
Hmm, I thought that in the case of no tx etc, there will be at least an empty root (the base case in [here](https://github.com/lazyledger/lazyledger-specs/blob/master/specs/data_structures.md#namespace-merkle-tree)). I'll think about this on Monday but these edge-cases should probably be more clearly addressed in the spec as well.

Reply:
Ah you're actually half-right. 0 is not a (non-negative, finite) power of 2, so the minimum original square size should be 1, not 0. So you'd have one row root and one column root, and they would be the root of a tree with a single zero-leaf. I'll clarify in the specs.

Reply:
@liamsi, @adlerjohn, is the convo resolved? Does the finding somehow affect the PR? Do I need to remove the method or change it to check for 1 length instead of 0?

Reply:
Do we need the method anyways?

Reply:
Right now - nobody. I kept it from other PR just in case, but let's keep the codebase clean without unnecessary things.

Reply:
Removed. We can add it further if needed

Reply:
No, better to remove it for now. We can figure it out when we see an actual need arise.